tests/escape.rs

```diff
@@ -0,0 +1,44 @@
+extern crate ron;
+extern crate serde;
+
+use std::char::from_u32;
+use std::fmt::Debug;
+
+use ron::de::from_str;
+use ron::ser::to_string;
+use serde::{Deserialize, Serialize};
+
+#[test]
+fn test_control_seq() {
+    assert_eq!(to_string(&"\x07").unwrap(), "\"\\x07\"");
+
+    assert_eq!(from_str::<String>("\"\\x07\"").unwrap(), "\x07");
+}
+
+fn check_same<T>(t: T)
+where
+    T: Debug + for<'a> Deserialize<'a> + PartialEq + Serialize,
+{
+    let s: String = to_string(&t).unwrap();
+
+    println!("Serialized: \n\n{}\n\n", s);
+
+    assert_eq!(from_str(&s), Ok(t));
+}
+
+#[test]
+fn test_ascii_10() {
+    check_same("\u{10}".to_owned());
```

Comment:
That's what causes the issue: the parser reads this as `\x01` instead of `\x10`.